lib/r10k/deployment/config.rb

```diff
@@ -30,7 +31,8 @@ def setting(key)
   def load_config
     loader = R10K::Settings::Loader.new
     hash = loader.read(@configfile)
-    @config = R10K::Settings.global_settings.evaluate(hash)
+    with_overrides = hash.merge(@overrides)
```

Comment:
I would like for us to log when config values are overridden; there's the block form of `Hash#merge` that could be used to log when overrides replace values - perhaps like so?

```ruby
with_overrides = hash.merge(@overrides) do |key, oldval, newval|
  logger.debug2 "Overriding config setting #{key} with old value #{oldval} to new value #{newval}"
end
```

I'm thinking that having this logged may make it easier to debug support requests.

Thoughts?